src/LiveComponent/src/Controller/BatchActionController.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\UX\LiveComponent\LiveComponentHydrator;
+use Symfony\UX\TwigComponent\ComponentFactory;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @internal
+ */
+final class BatchActionController
+{
+    public function __construct(
+        private HttpKernelInterface $kernel,
+        private ComponentFactory $factory,
+        private LiveComponentHydrator $hydrator,
+    ) {
+    }
+
+    public function __invoke(Request $request, string $component): Response
+    {
+        try {
+            $json = json_decode($request->getContent(), true, 512, \JSON_THROW_ON_ERROR);
+        } catch (\JsonException) {
+            throw new BadRequestHttpException('Invalid JSON');
+        }
+
+        $data = $json['data'] ?? throw new BadRequestHttpException('Invalid JSON');
+        $actions = $json['actions'] ?? throw new BadRequestHttpException('Invalid JSON');
+
+        if (!\is_array($actions) || !$actions) {
+            throw new BadRequestHttpException('Invalid JSON');
+        }
+
+        $componentObject = $this->factory->get($component);
+        $mountedComponent = $this->hydrator->hydrate($componentObject, $data, $component);
```

Comment:
I was expecting all the stuff above this would *still* live in `LiveComponentSubscriber` - and then this controller is called (instead of `__invoke()` on the component or a real action method) only to handle the action execution.